charts/backstage/templates/_helpers.tpl

```diff
@@ -5,6 +5,17 @@ Return the proper image name
 {{ include "common.images.image" (dict "imageRoot" .Values.backstage.image "global" .Values.global) }}
 {{- end -}}
 
+{{/*
+ Create the name of the service account to use
+ */}}
+{{- define "backstage.serviceAccountName" -}}
+{{- if .Values.serviceAccount.create -}}
+    {{ default (include "common.names.fullname" .) .Values.serviceAccount.name }}
+{{- else -}}
+    {{ default "default" .Values.serviceAccount.name }}
```

Comment:
So from a security perspective here, default if not otherwise defined is what would be used for all pods in the given namespace. While we don't associate the role with any permisisons yet, can this have unintentional impact of providing unintended access by poor configuration by others. Should we not use "default" in the particular case and create a backstage based name as the default?

Reply:
To clarify, not specifying `serviceAccountName` in the Pod means that the [`default` service account to be used](https://github.com/kubernetes/kubernetes/blob/407bd6a4afa5f7d7cb2a596dda09cacc30828f09/plugin/pkg/admission/serviceaccount/admission.go#L45).

This PR doesn't change the current behavior from a security perspective.

In my opinion, having a backstage service account created by default is not related to this PR and should be handled separately if wanted.

Reply:
As pointed out by @spasche, if you submit a Deployment/Pod manifest without `spec.serviceAccountName` defined, it gets reconciled to `default` value anyways. This is a hardcoded value in kubernetes.

While it may be better to leave this field blank instead of hardcoding in the `default` value, this pattern is so widely used in other helm charts, that I don't see any point in nit picking that.